hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -269,36 +270,145 @@ public void start() {
         .values()
         .forEach(ReconOmTask::init);
 
-    // Verify if 'OmDeltaRequest' task's lastUpdatedSeqNumber number not matching with
+    // Verify if 'OmDeltaRequest' task's lastUpdatedSeqNumber is greater than zero and greater than
+    // 'OmSnapshotRequest' task's lastUpdatedSeqNumber number and not matching with
     // lastUpdatedSeqNumber number for any of the OM task, then just run reprocess for such tasks.
+
+    ReconTaskStatusUpdater fullSnapshotTaskStatusUpdater =
```

Comment:
Why have a new fullSnapshotTaskStatusUpdater ? We can update the same variable irrespective of it being a delta process or reprocess.
We just have to bootstrap if there is difference between omSnapshot & OM Leader sequence number and provided all TaskStatusUpdater tasks = OmSnapshot.sequenceNumber. On end of every process or reprocess we should just update the sequenceNumber in taskStatusUpdater to the OmSnapshot's sequence number. If it doesn't match we should just run reprocess if the value is 0 or rerun process from the seekPos. Isn't this possible?